# origin: Debian, #155570
# cset: n/a
# inclusion: not submitted
# description: make modular binfmt_misc unloadable
# revision date: 2004-09-04


diff -urN kernel-source-2.4.26/fs/binfmt_misc.c kernel-source-2.4.26-1/fs/binfmt_misc.c
--- kernel-source-2.4.26/fs/binfmt_misc.c	2002-08-03 10:39:45.000000000 +1000
+++ kernel-source-2.4.26-1/fs/binfmt_misc.c	2003-05-18 17:52:06.000000000 +1000
@@ -682,22 +682,13 @@
 
 static DECLARE_FSTYPE(bm_fs_type, "binfmt_misc", bm_read_super, FS_SINGLE|FS_LITTER);
 
-static struct vfsmount *bm_mnt;
-
 static int __init init_misc_binfmt(void)
 {
 	int err = register_filesystem(&bm_fs_type);
 	if (!err) {
-		bm_mnt = kern_mount(&bm_fs_type);
-		err = PTR_ERR(bm_mnt);
-		if (IS_ERR(bm_mnt))
+		err = register_binfmt(&misc_format);
+		if (err) {
 			unregister_filesystem(&bm_fs_type);
-		else {
-			err = register_binfmt(&misc_format);
-			if (err) {
-				unregister_filesystem(&bm_fs_type);
-				kern_umount(bm_mnt);
-			}
 		}
 	}
 	return err;
@@ -707,7 +698,6 @@
 {
 	unregister_binfmt(&misc_format);
 	unregister_filesystem(&bm_fs_type);
-	kern_umount(bm_mnt);
 }
 
 EXPORT_NO_SYMBOLS;
